### PR TITLE
update to Gnome 40

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by evfool** [comment]
Hi, thanks for the PR, this would look fine, however it looks like appdata validate fails. I need to check whether the problem is that the appdata contains the latest version first (adding to the top always), or the new versioning scheme is not working out fine with appdata.

**Comment by evfool** [comment]
Accorting to https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases the release ordering is correct, quoting the spec "The release children should be sorted in a latest-to-oldest order to simplify reading the metadata file. ", so the new versioning scheme being in conflict with appdata-validate is the problem here.

**Comment by Lctrs** [comment]
Yep, there is a bug report for that : https://github.com/hughsie/appstream-glib/issues/391

We should probably disable it temporarily.

**Comment by Lctrs** [comment]
@barthalion Any advice please ?

**Comment by bilelmoussaoui** [comment]
> @barthalion Any advice please ?

You should probably patch out the beta/rc entries from the appdata file.

File: org.gnome.Calculator.json [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
-    "runtime-version": "3.38",
+    "runtime-version": "40",
     "sdk": "org.gnome.Sdk",
     "command": "gnome-calculator",
     "finish-args": [
-        /* X11 + XShm access */
-        "--share=ipc", "--socket=fallback-x11",
-        /* Wayland access */
+        "--share=ipc",
+        "--socket=x11",
```
Comment (bilelmoussaoui) [inline org.gnome.Calculator.json:9]: Why x11 instead of fallback-x11?

Comment (Lctrs) [reply org.gnome.Calculator.json:9]: Whoops. Bad copy/paste from upstream. Didn't see this one.

**Comment by Lctrs** [comment]
> > @barthalion Any advice please ?
> 
> You should probably patch out the beta/rc entries from the appdata file.

@bilelmoussaoui I just applied the same trick as in https://gitlab.gnome.org/GNOME/epiphany/-/commit/81ca16fcfd434faaa0a15cb28901294b86b681f8

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by evfool** [comment]
@bilelmoussaoui  @Lctrs regarding x11 vs fallback-x11, should I be using fallback-x11 also upstream?

**Comment by bilelmoussaoui** [comment]
> @bilelmoussaoui @Lctrs regarding x11 vs fallback-x11, should I be using fallback-x11 also upstream?

Yes, `fallback-x11` gives the application x11 socket access only as a fallback to the wayland one instead of always giving it access to x11.

**Comment by evfool** [comment]
Thanks, updated upstream flatpak json accordingly.

**Pull request merged**